src/Quizzer.API.BL/Facades/QuizFacade.cs

```diff
@@ -0,0 +1,167 @@
+                InProgress = true
+            });
+
+            uow.Commit();
+        }
+
+        public void CompleteQuiz(Guid quizId, Guid userId) {
+            using IUnitOfWork uow = p_unitOfWorkFactory.Create();
+            using IRepository<CompletedQuizEntity> completedQuizRepository = uow.GetRepository<CompletedQuizEntity>();
+
+            CompletedQuizEntity? completedQuiz = completedQuizRepository.GetByPredicate(
+                q => q.QuizId == quizId && q.UserId == userId,
+                nameof(CompletedQuizEntity.Quiz), nameof(CompletedQuizEntity.User)).SingleOrDefault()
+                ?? throw new InvalidOperationException("Quiz has not been started");
+
+            if (!completedQuiz.InProgress) throw new InvalidOperationException("Quiz has been completed");
+
+            completedQuiz.InProgress = false;
+
+            using IRepository<UserAnswerEntity> userAnswerRepository = uow.GetRepository<UserAnswerEntity>();
+            IEnumerable<UserAnswerEntity> answers = userAnswerRepository.GetByPredicate(
+                a => a.UserId == userId && a.Question!.QuizId == quizId,
+                nameof(UserAnswerEntity.Question), $"{nameof(UserAnswerEntity.Question)}.{nameof(QuestionEntity.Quiz)}");
+
+            completedQuiz.Score = answers.Select(a => a.AnswerScore).Sum();
+            completedQuizRepository.Update(completedQuiz);
+
+            uow.Commit();
+        }
+
+        public void AddQuestion(Guid questionId, Guid quizId, DateTime time) {
+            using IUnitOfWork uow = p_unitOfWorkFactory.Create();
+            using (IRepository<QuestionEntity> questionRepository = uow.GetRepository<QuestionEntity>()) {
+                using (IRepository<QuizEntity> quizRepository = uow.GetRepository<QuizEntity>()) {
+                    if (!quizRepository.Exists(quizId)) throw new ArgumentException("Quiz does not exist");
+                }
+
+                QuestionEntity? questionEntity = questionRepository.GetById(questionId)
+                    ?? throw new ArgumentException("Question does not exist");
+
+                questionEntity.QuizId = quizId;
+
+                questionRepository.Update(questionEntity);
+            }
+            uow.Commit();
+        }
+
+        public QuizResultDetailModel GetResultModel(Guid quizId, Guid userId, DateTime time) {
+            using IUnitOfWork uow = p_unitOfWorkFactory.Create();
+            
+            using IRepository<QuizEntity> quizRepository = uow.GetRepository<QuizEntity>();
+
+            QuizEntity? quizEntity = quizRepository.GetById(quizId, nameof(QuizEntity.CompletedQuizzes));
+            CompletedQuizEntity? quizResult = quizEntity.CompletedQuizzes.Where(q => q.UserId == userId).FirstOrDefault();
+
+            if (quizResult is null || quizResult.InProgress) throw new ArgumentException("Quiz was not completed");
+
+            if (quizEntity.EndTime > time) throw new InvalidOperationException("Quiz is still open");
```

Comment:
So when the quiz is opened you cant see your results?

Reply:
Because there could be someone who does not completed the quiz, thus the Max score could not be definite